Source/WebCore/platform/graphics/coretext/FontPlatformDataCoreText.cpp

```diff
@@ -46,6 +66,19 @@ FontPlatformData::FontPlatformData(RetainPtr<CTFontRef>&& font, float size, bool
 #if PLATFORM(IOS_FAMILY)
     m_isEmoji = CTFontIsAppleColorEmoji(m_font.get());
 #endif
+
+    if (m_widthVariant != FontWidthVariant::RegularWidth) {
+        int featureTypeValue = kTextSpacingType;
+        int featureSelectorValue = mapFontWidthVariantToCTFeatureSelector(m_widthVariant);
+        RetainPtr<CTFontDescriptorRef> sourceDescriptor = adoptCF(CTFontCopyFontDescriptor(m_font.get()));
+        RetainPtr<CFNumberRef> featureType = adoptCF(CFNumberCreate(kCFAllocatorDefault, kCFNumberIntType, &featureTypeValue));
+        RetainPtr<CFNumberRef> featureSelector = adoptCF(CFNumberCreate(kCFAllocatorDefault, kCFNumberIntType, &featureSelectorValue));
+        RetainPtr<CTFontDescriptorRef> newDescriptor = adoptCF(CTFontDescriptorCreateCopyWithFeature(sourceDescriptor.get(), featureType.get(), featureSelector.get()));
+        RetainPtr<CTFontRef> newFont = adoptCF(CTFontCreateWithFontDescriptor(newDescriptor.get(), m_size, 0));
+
+        if (newFont)
+            m_font = newFont;
```

Comment:
I wonder if there is a way to do this before we instantiate m_font for the first time (when we create FontPlatformData) so we do an extra `CTFontCreateWithFontDescriptor`

Reply:
I thought about that, and I agree that long-term that's what we should do, but it would complicate this patch a lot and I didn't want to make those changes. This works, and it's not a regression from what we've been doing.

Reply:
Sounds reasonable but then let's create a radar with a FIXME here so we don't forget about it.

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=276635